openeo_processes_dask/process_implementations/cubes/load.py

```diff
@@ -0,0 +1,164 @@
+
+import planetary_computer as pc
+import pyproj
+import pystac_client
+import stackstac
+import xarray as xr
+from openeo_pg_parser_networkx.pg_schema import *
+from stac_validator import stac_validator
+
+from openeo_processes_dask.process_implementations.cubes._filter import (
+    _reproject_bbox,
+    filter_bands,
+    filter_bbox,
+    filter_temporal,
+)
+from openeo_processes_dask.process_implementations.data_model import RasterCube
+from openeo_processes_dask.process_implementations.exceptions import (
+    NoDataAvailable,
+    TemporalExtentEmpty,
+)
+
+# "NoDataAvailable": {
+#     "message": "There is no data available for the given extents."
+# },
+# "TemporalExtentEmpty": {
+#     "message": "The temporal extent is empty. The second instant in time must always be greater/later than the first instant in time."
+# }
+__all__ = ["load_stac"]
+
+
+def load_stac(
+    url: str,
+    spatial_extent: BoundingBox = None,
+    temporal_extent: Optional[
+        list[Union[str, datetime.datetime, datetime.date]]
+    ] = None,
+    bands: Optional[list[str]] = None,
+    properties: Optional[dict] = None,
+    **kwargs,
+) -> RasterCube:
+    def validate_stac(url):
+        stac = stac_validator.StacValidate(url)
+        is_valid_stac = stac.run()
+        if not is_valid_stac:
+            raise Exception(
+                f"The provided link is not a valid STAC. stac-validator message: {stac.message}"
+            )
+        if len(stac.message) == 1:
+            try:
+                asset_type = stac.message[0]["asset_type"]
+            except:
+                raise Exception(f"stac-validator returned an error: {stac.message}")
+        else:
+            raise Exception(
+                f"stac-validator returned multiple items, not supported yet. {stac.message}"
+            )
+        return is_valid_stac, asset_type
+
+    is_valid_stac, asset_type = validate_stac(url)
+
+    if not is_valid_stac:
+        raise Exception("The provided link is not a valid STAC.")
+
+    if asset_type == "COLLECTION":
+        # If query parameters are passed, try to get the parent Catalog if possible/exists, to use the /search endpoint
+        if spatial_extent or temporal_extent or bands or properties:
+            # If query parameters are passed, try to get the parent Catalog if possible/exists, to use the /search endpoint
+            # url = "https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs"
+            parsed_url = urlparse(url)
+            root_url = parsed_url.scheme + "://" + parsed_url.netloc
+            catalog_url = root_url
+            url_parts = PurePosixPath(unquote(parsed_url.path)).parts
+            collection_id = url_parts[-1]
+            for p in url_parts:
+                if p != "/":
+                    catalog_url = catalog_url + "/" + p
+                try:
+                    is_valid_stac, asset_type = validate_stac(catalog_url)
```

Comment:
This is pretty confusing: `asset_type` (and `is_valid_stac`) are overwritten here. is that intentional?

Reply:
Yes, it is intentional, since from a STAC Collection url I look for the root Catalog url, where I can perform a query and obtain the items to load. Maybe it's not the best approach, I will try to get in contact with the pystac_client developers to see if there's a better solution.